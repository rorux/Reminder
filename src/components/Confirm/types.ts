import React from 'react';

export type TConfirmProps = {
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  handler: (event: React.MouseEvent<HTMLElement>, id: string) => void;
  id: string;
};
