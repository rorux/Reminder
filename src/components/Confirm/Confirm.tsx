import React, { useEffect } from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import { TConfirmProps } from './types';

const Confirm: React.FC<TConfirmProps> = ({ open, setOpen, handler, id }) => {
  const [openConfirm, setOpenConfirm] = React.useState(true);

  useEffect(() => {
    setOpenConfirm(open);
  }, [open]);

  const handleClose = () => {
    setOpen(false);
  };

  const handlerFunc = (e: React.MouseEvent<HTMLElement>, id: string) => {
    handler(e, id);
    setOpen(false);
  };

  return (
    <>
      <Dialog
        open={openConfirm}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogContent>
          <DialogContentText>Удалить напоминание?</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button data-test="disagree" onClick={handleClose}>
            Отмена
          </Button>
          <Button data-test="agree" onClick={(e) => handlerFunc(e, id)} autoFocus>
            Удалить
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default Confirm;
