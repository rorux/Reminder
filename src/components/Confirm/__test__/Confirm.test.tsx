import * as React from 'react';
import * as Redux from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import Confirm from '../Confirm';

Enzyme.configure({ adapter: new Adapter() });

const setUp = () =>
  shallow(<Confirm open={false} setOpen={jest.fn()} handler={jest.fn()} id={'1s3'} />);

describe('Confirm component', () => {
  let component: Enzyme.ShallowWrapper;
  beforeEach(() => {
    component = setUp();
  });

  it('Snapshot of Confirm component', () => {
    expect(component).toMatchSnapshot();
  });
});
