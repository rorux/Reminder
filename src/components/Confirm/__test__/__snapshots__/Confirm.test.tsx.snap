// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm component Snapshot of Confirm component 1`] = `
<Fragment>
  <ForwardRef(Dialog)
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    onClose={[Function]}
    open={true}
  >
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText)>
        Удалить напоминание?
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        data-test="disagree"
        onClick={[Function]}
      >
        Отмена
      </ForwardRef(Button)>
      <ForwardRef(Button)
        autoFocus={true}
        data-test="agree"
        onClick={[Function]}
      >
        Удалить
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</Fragment>
`;
